lld/COFF/Driver.cpp

```diff
@@ -1039,12 +1040,14 @@ void LinkerDriver::parseModuleDefs(StringRef path) {
     if (!e1.ExtName.empty() && e1.ExtName != e1.Name &&
         StringRef(e1.Name).contains('.')) {
       e2.name = saver().save(e1.ExtName);
+      e2.exportAs = saver().save(e1.ExportAs);
```

Comment:
This doesn't seem to match what the comment above is describing - is `EXPORTAS` allowed in this "simple case"?

Reply:
It is allowed with MSVC linker. It's the same preexisting problem as in #86535, but when using .def files. I created #86564 with a fix for that (which allows skipping this part as it no longer uses a separated branch). I rebased this PR on top of that.